### DISCUSSION TITLE
FJM - RJ100F5HXBA - MultiSplit CRC error NonNasa

### DISCUSSION THREAD

**Discussion by Goaheadz** [omerfaruk-aran/esphome_samsung_hvac_bus#72] [Questions]
Using ESP8266 D1 Mini connected to a UART->RS485 board.
Cant use TX and RX pins from D1 mini, cause somehow, connecting RX from D1 mini to RX from TTL-RS485, turns led off (killing transmission) from board...tried 2 diferent D1 mini and rs485 pcbs..same problem.

I have to use GPIO13 and GPIO15 (SW UART)
Dont know if it a HW or SW problem but I get crc errors,. Outdoor, indoor splits and outdoor sensor board are being detected, but nothing works when I send power on :(

![image](https://github.com/lanwin/esphome_samsung_ac/assets/13796554/9ebc9560-2b2f-498b-b212-988a9c0651bf)

Is the CRC error (nonNasa) something you can help?

Regards

**Comment by Goaheadz** [comment]
C8 must be outdoor mainboard and 04 and CC other board from the outdoor unit.
Tried all combinations possible.
- interchange F1 and F2
- RX and TX
- Baud-rate from 1200 to 19200
- Parity
- Different D1 and RS485 boards

More than 8 hours into this prject,  and no luck so far

**Comment by Goaheadz** [comment]
**"Cant use TX and RX pins from D1 mini, cause somehow, connecting RX from D1 mini to RX from TTL-RS485, turns led off (killing transmission) from board...tried 2 diferent D1 mini and rs485 pcbs..same problem."**

For this issue, I think 3.3v is getting to low. Small LDO from D1 mini may not supply suficient current/voltage and the transceiver may disconnect.
I have to get nother 3.3v supply and try this again.

![image](https://github.com/lanwin/esphome_samsung_ac/assets/13796554/a41be78e-e0a1-4f66-8a14-ddd176bca1c5)

Dont know if crc error is linkd to this.

just guessing.

**Comment by Goaheadz** [comment]
Finally some progress.
RX side is working.
Probably was working from day 1, but never notice that when I power on AC with remote, it would update data received in rs485 data and HA.
![image](https://github.com/lanwin/esphome_samsung_ac/assets/13796554/df4a5098-bd7a-437d-a21a-764fb849f1c1)

TX is not working thought.
Dont see led blink in UART-RS485 neither any special code in log window
Let me log only TX data now, to make it less confusing.

Half way there :)

**Comment by Goaheadz** [comment]
`  debug:
    direction: TX
    dummy_receiver: false
    after:
      delimiter: "\n"
    sequence:
      - lambda: UARTDebug::log_string(direction, bytes);`
      
My AC splits dont have lcd indicator..just power led.
**No data can be seen, being sent in logs, when buttons are pressed in HA.(I mean rs485 data)**
Only changing temperature in HA stays with selected temperature after pressed. But like said above..splits dont have lcd to confirm.
If I change fan speed or any other feature, it goes back to current state. No changes can be made in HA.

**Comment by lanwin** [comment]
Hi, wow. As of right now the CRC errors are normal. A tiny part of the messages produces them but since this also happens on NASA devices I would guess this is normal.

The Non NASA part is currently no 100% functional. I am made some huge progress the last days but there are still some gaps to find out. Please follow this thread here https://github.com/lanwin/esphome_samsung_ac/issues/1

**Reply by Goaheadz** [reply]
@lanwin Thank you for all your time and effort into this.

I will keep following the other nonNasa tread and reply to it if I get progress..

Currently trying to be sure that TX is working on my side (esp8266->TTL->RS485), as I dont see any led blinking when transmitting.

Available to any type of testing you may need.

**Reply by lanwin** [reply]
Do you see a write message when changing anything? If yes then a message should get sent.

**Reply by Goaheadz** [reply]
Topic can be closed.
TX is now working.
